### PR TITLE
Introduce react hook form macaw bindings

### PR DESCRIPTION
**Pull request by krzysztofwolski**
New shared package with Macaw components ready to use with React Hook Form.
Other libraries used as reference: https://react-hook-form.com/resources/3rd-party-bindings/

In short - instead of using `Controller` from RHF to wrap input, use Input from the shared package and cut boilerplate in half :) 
Example of usage: https://github.com/saleor/apps/pull/500

### CODE REVIEW

File: packages/react-hook-form-macaw/.eslintrc
Comment (lkostrowski) [inline packages/react-hook-form-macaw/.eslintrc:1]: maybe we can rename pkg to `macaw-rhf` but no strong opinion

File: packages/react-hook-form-macaw/.storybook/main.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { mergeConfig } from "vite";
+import { resolve } from "path";
+
+export default {
```
Comment (lkostrowski) [inline packages/react-hook-form-macaw/.storybook/main.ts:4]: Please add issue to consider extracting storybook to shared package that will be playground for this and other packages

File: packages/react-hook-form-macaw/.storybook/preview.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  return (
+    <ThemeProvider defaultTheme={props.context.store.globals.globals.theme}>
+      <ThemeSwitcher theme={props.context.store.globals.globals.theme}>
+        <DocsContainer {...props}>{children}</DocsContainer>
+      </ThemeSwitcher>
+    </ThemeProvider>
+  );
+};
+
+export const parameters = {
```
Comment (lkostrowski) [inline packages/react-hook-form-macaw/.storybook/preview.tsx:25]: please make these values to `satisfies` or have assigned types, so we can safely edit them later, I assume storybook has typings

**Review by lkostrowski — CHANGES_REQUESTED** [review]

File: packages/react-hook-form-macaw/.eslintrc
Comment (krzysztofwolski) [reply packages/react-hook-form-macaw/.eslintrc:1]: Would rather follow the naming scheme of other binding libs

**Review by lkostrowski — APPROVED** [review]

**Review by lkostrowski — APPROVED** [review]

File: packages/react-hook-form-macaw/.storybook/main.ts (hunk 1 above)
Comment (krzysztofwolski) [reply packages/react-hook-form-macaw/.storybook/main.ts:3]: https://github.com/saleor/apps/issues/501 Added :+1:

File: packages/react-hook-form-macaw/.storybook/preview.tsx (hunk 2 above)
Comment (krzysztofwolski) [reply packages/react-hook-form-macaw/.storybook/preview.tsx]: Improved types as much as I could. Fell into the rabbithole of SB generics and there are known issues with it.

**Pull request merged**